numpy/tests/typing/fail/flatiter.py

```diff
@@ -0,0 +1,21 @@
+from typing import Any
+
+import numpy as np
+from numpy.typing import DtypeLike, _SupportsArray
+
+
+class Index:
+    def __index__(self) -> int:
+        ...
+
+
+a: "np.flatiter[np.ndarray]"
+supports_array: _SupportsArray
+
+a.base = Any  # E: Property "base" defined in "flatiter" is read-only
+a.coords = Any  # E: Property "coords" defined in "flatiter" is read-only
+a.index = Any  # E: Property "index" defined in "flatiter" is read-only
+a.copy(order='C')  # E: Unexpected keyword argument
+a[np.bool_()]  # E: No overload variant of "__getitem__"
+a[Index()]  # E: No overload variant of "__getitem__"
```

Comment:
Shouldn't this be legal? Might be worth referencing an issue if its not.

Reply:
It should be, but the behavior of `flatiter.__getitem__` is very much inconsistent 
with respect to `ndarray.__getitem__` (case in point: https://github.com/numpy/numpy/issues/17175).

Adding a note is probably a good idea though (see https://github.com/numpy/numpy/pull/17180/commits/edcf0ea581546eb8c29ba34b7576f2a8d81b4eec).